src/main/java/pl/psnc/dei/controllers/ImportController.java

```diff
@@ -22,32 +22,41 @@ public ImportController(ImportPackageService importService) {
         this.importService = importService;
     }
 
-    @PostMapping("import/create/projectId/{projectId}")
-    public ResponseEntity<Import> createImport(@PathVariable("projectId") String projectId, @RequestParam(value = "name", required = false) String name, @RequestBody List<Record> records) {
+    @PostMapping("import/create")
+    public ResponseEntity<Import> createImport(@RequestParam(value = "projectId") String projectId, @RequestParam(value = "name", required = false) String name, @RequestBody List<Record> records) {
         return new ResponseEntity<>(importService.createImport(name, projectId, records), HttpStatus.OK);
     }
 
-    @GetMapping("/import/candidates/projectId/{projectId}")
-    public ResponseEntity<List<Record>> getCandidates(@PathVariable String projectId, @RequestParam(value = "datasetId", required = false) String datasetId) {
+    @GetMapping("/import/candidates")
+    public ResponseEntity<List<Record>> getCandidates(@RequestParam(value = "projectId") String projectId, @RequestParam(value = "datasetId", required = false) String datasetId) {
         return new ResponseEntity<>(importService.getCandidates(projectId, datasetId), HttpStatus.OK);
     }
 
-    @GetMapping("/import/status/{importName}")
-    public ResponseEntity getImportReport(@PathVariable String importName) {
+    @GetMapping("/import/status")
+    public ResponseEntity getImportReport(@RequestParam(value = "importName") String importName) {
         try {
             return new ResponseEntity<>(importService.getStatusWithFailure(importName), HttpStatus.OK);
         } catch (NotFoundException e) {
             return new ResponseEntity<>(HttpStatus.NOT_FOUND);
         }
     }
 
-    @PostMapping("/import/send/{importName}")
-    public ResponseEntity sendImport(@PathVariable String importName) {
+    @PostMapping("/import/send")
+    public ResponseEntity sendImport(@RequestParam(value = "importName") String importName) {
         try {
             importService.sendExistingImport(importName);
             return new ResponseEntity<>(HttpStatus.OK);
         } catch (NotFoundException e) {
             return new ResponseEntity<>(HttpStatus.NOT_FOUND);
         }
     }
+
+    @PutMapping("/import")
```

Comment:
If we want to be consistent this endpoint should be maybe "/import/update", what do you think?

Reply:
We can but I think general convention is not to use verbs in endpoints